### PR TITLE
[Docs][Kuberay] Create an example doc for Modin

### LINKED ISSUE #2190: [Feature] Create an example doc for Modin
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

https://ray-distributed.slack.com/archives/C02GFQ82JPM/p1718067374071799?thread_ts=1717741098.825609&cid=C02GFQ82JPM

Add a doc under https://docs.ray.io/en/latest/cluster/kubernetes/examples.html

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by MortalHappiness**
## Why are these changes needed?

See the description in the corresponding issue for details.
Related PR: https://github.com/ray-project/kuberay/pull/2221

## Related issue number

Resolves: ray-project/kuberay#2190

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by MortalHappiness** [comment]
@kevin85421 Could you help me review this PR? Thanks.

File: doc/source/cluster/kubernetes/examples/modin-example.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+## Step 1: Install Kuberay operator
+
+Follow the steps 1 and 2 from the [RayCluster Quick Start](kuberay-raycluster-quickstart) guide to install Kuberay operator.
+
+## Step 2: Run the Modin example with RayJob
+
+Create a file named `ray-job.modin.yaml` with the following content:
+
+```yaml
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/examples/modin-example.md:13]: Add a YAML KubeRay repository.

File: doc/source/cluster/kubernetes/examples/modin-example.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                    cpu: "1"
+                  requests:
+                    cpu: "200m"
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ray-job-code-sample
+data:
+  sample_code.py: |
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/examples/modin-example.md:81]: Use a real-world example instead.

**Review by can-anyscale — CHANGES_REQUESTED** [review]
- push back to your queue for @kevin85421's feedback
- do you mind add a screenshot or link to the generated page, thankks

**Comment by MortalHappiness** [comment]
@can-anyscale PTAL. Thanks.
image

File: doc/source/cluster/kubernetes/examples/modin-example.md (hunk 1 above)
Comment (MortalHappiness) [reply doc/source/cluster/kubernetes/examples/modin-example.md]: Done

File: doc/source/cluster/kubernetes/examples/modin-example.md (hunk 2 above)
Comment (MortalHappiness) [reply doc/source/cluster/kubernetes/examples/modin-example.md]: Done

File: doc/source/cluster/kubernetes/examples/modin-example.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/ray-job.modin.yaml
+```
+
+## Step 3: Check the output
+
+Run the following command to check the output:
+
+```sh
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/examples/modin-example.md:21]: Have you ever updated the logs after unwrapping the Modin example from a Ray task?

Comment (MortalHappiness) [reply doc/source/cluster/kubernetes/examples/modin-example.md:21]: Done

**Review by angelinalg — APPROVED** [review]
Some suggestions for style consistency. Please consider installing Vale to check for these corrections. https://docs.ray.io/en/latest/ray-contribute/docs.html#how-to-use-vale

Please also verify the technical accuracy of my suggestion for an introductory sentence for Step 2, proposed line 13.

Thanks for making Ray docs better!!

File: doc/source/cluster/kubernetes/examples/modin-example.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+# Use Modin with Ray on Kubernetes
+
+In this example, you will learn how to run a modified version of the [Using Modin with the NYC Taxi Dataset](https://github.com/modin-project/modin/blob/4e7afa7ea59c7a160ed504f39652ff23b4d49be3/examples/jupyter/Modin_Taxi.ipynb) example from the Modin official repository using RayJob on Kubernetes.
+
+## Step 1: Install Kuberay operator
+
+Follow the steps 1 and 2 from the [RayCluster Quick Start](kuberay-raycluster-quickstart) guide to install Kuberay operator.
+
+## Step 2: Run the Modin example with RayJob
+
```
[the remark this answers is not in the document]
Comment (MortalHappiness) [reply doc/source/cluster/kubernetes/examples/modin-example.md:12]: @angelinalg Thank you! I've pushed updates according to your suggestions.

**Review by kevin85421 — APPROVED** [review]

**Review by can-anyscale — APPROVED** [review]

**Pull request merged**